### PR TITLE
BUILD-3017 use tokens for JFrog Artifactory 

### PR DESCRIPTION
**Pull request by malena-ebert-sonarsource**

### CODE REVIEW

**Review by julien-carsique-sonarsource — APPROVED** [review]

File: azure-pipelines.yml [hunk 1]
```diff
@@ -118,13 +134,23 @@ stages:
...
-          artifactoryService: 'Repox artifactory'
+          artifactoryConnection: 'repox_qa_deployer_token'
           targetDeployRepo: '$(ARTIFACTORY_NUGET_REPO)'
+          module: SonarAnalyzer.CFG
           pathToNupkg: '$(Build.ArtifactStagingDirectory)/packages/SonarAnalyzer.CFG.*.nupkg'
+          collectBuildInfo: true
+          buildName: sonar-dotnet_nuget
+          buildNumber: '$(Build.BuildId)'
+
+      - task: JFrogPublishBuildInfo@1
```
Comment (andrei-epure-sonarsource) [inline azure-pipelines.yml:149]: nice catch

File: azure-pipelines.yml [hunk 2]
```diff
@@ -540,52 +550,20 @@ stages:
...
-          env:
-            ARTIFACTORY_PROMOTER_ACCESS_TOKEN: $(ARTIFACTORY_PROMOTER_ACCESS_TOKEN)
+            artifactoryConnection: 'repox_promoter_token'
+            buildName: sonar-dotnet_nuget
+            buildNumber: '$(Build.BuildId)'
+            sourceRepo: '$(ARTIFACTORY_NUGET_REPO)'
+            ${{ if eq(variables['Build.Reason'], 'PullRequest') }}:
+              targetRepo: sonarsource-nuget-dev
+            ${{ else }}:
+              targetRepo: sonarsource-nuget-public
```
[the remark this answers is not in the document]
Comment (andrei-epure-sonarsource) [reply azure-pipelines.yml:564]: in other words, is https://repox.jfrog.io/repox/api/nuget/sonarsource-nuget-releases publicly available?

**Review by andrei-epure-sonarsource — CHANGES_REQUESTED** [review]
I left a question on the nuget repox being public or not (just a double check)

File: azure-pipelines.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (andrei-epure-sonarsource) [reply azure-pipelines.yml:564]: There shouldn't be a big problem in being public, as it comes from an open-source repo, I am just checking the existing bevahior.

[the remark this answers is not in the document]
Comment (andrei-epure-sonarsource) [reply azure-pipelines.yml:564]: Thanks - it's here https://repox.jfrog.io/ui/native/sonarsource-nuget-releases/

So it's public. It's ok, I just wanted to double check. 

Sorry for the noise.

**Review by andrei-epure-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**